Feature/cheyenne

natalie-perlin opened the pull request:
Creating a pull request similar to the #444 , to make the build of the HPC-stack on Cheyenne and prevent system default modules overloading. This uses Lmod functionality of using "module collections"

 It uses `module save <lmod_collection>` and `module restore <lmod_collection>` commands, and the collections of modules are saved in user's home directory `~/.lmod.d/.` This is a standard way similar to user space configuration using ~/.ssh/ directories, or ~/.matlab/ , ~/.vim, ~/.gitconfig .
An example of such a collection, the file ~/.lmod.d/hpc-mpi-esmf.ch on cheyenne in `nperlin` home directory, is attached, except with the name modified to the allowed format to attach it, with the added *.TXT extension.

Additional functionality is used to prevent the system to load default module versions, when a specific version of the same software module is needed and it is already loaded. The syntax uses 'module is-loaded' construct. A couple of examples:
module is-loaded hdf5 || module load hdf5
module is-loaded cmake || module try-load cmake
module is-loaded netcdf || module load netcdf

[hpc-mpi-esmf.ch.TXT](https://github.com/NOAA-EMC/hpc-stack/files/8637652/hpc-mpi-esmf.ch.TXT)

Adding log files from cheyenne's build that used `module restore` from Lmod collections.
Built with gnu/11.2.0, mpi/2.25.
HPC-stack contents: hpc-stack-contents_cheyenne_gnu11.2.log
Log file in the building directory: build_stack_cheyenne_gnu11.2.log
Compressed archived log/ directory with individual log files: log_cheyenne_gnu11.2.tgz (please note that it has been renamed to log_cheyenne_gnu11.2.tgz.zip to comply with naming formats allowed to be attached to the comment).
Notifications on modules being restored and module list following a "module restore" command could be found in log files, by searching strings *"Restoring modules ".

[hpc-stack-contents_cheyenne_gnu11.2.log](https://github.com/NOAA-EMC/hpc-stack/files/8701761/hpc-stack-contents_cheyenne_gnu11.2.log)
[build_stack_cheyenne_gnu11.2.log](https://github.com/NOAA-EMC/hpc-stack/files/8701770/build_stack_cheyenne_gnu11.2.log)
[log_cheyenne_gnu11.2.tgz.zip](https://github.com/NOAA-EMC/hpc-stack/files/8701813/log_cheyenne_gnu11.2.tgz.zip)

kgerheiser committed:
Fix BUILD_TESTS for bufr

Change TETS to TESTS

kgerheiser committed:
Merge pull request #372 from NOAA-EMC/feature/cheyenne

Fixes a few issues that were breaking the full build of stack_ufs_weather_ci.yaml on Cheyenne.

kgerheiser committed:
Merge branch 'develop' into bugfix/bufr-tests

Hang-Lei-NOAA committed:
part of issue #373"

kgerheiser committed:
Merge pull request #375 from Hang-Lei-NOAA/crtm_v2.4.0

Update CRTM to v2.4.0 which comes from a different remote.

aerorahul committed:
Add `csh` support for loading miniconda3 (#377)

kgerheiser committed:
Merge branch 'develop' into bugfix/bufr-tests

kgerheiser committed:
Merge pull request #368 from NOAA-EMC/bugfix/bufr-tests

Fix BUILD_TESTS for bufr

kgerheiser committed:
Skip build instead of exiting with failure if library exists (#364)

Skip build instead of exiting with failure if library exists

kgerheiser committed:
Re-name png to libpng (#389)

Make name consistent with project name and Spack install

kgerheiser committed:
Add stack_gfsv16.2.yaml and module variables (#392)

* Fix duplicate symbols in NCO

* Add stack_gfsv16.2.yaml

* Update modulefiles to support gfsv16.2

* Fix typo

* Update NCO version

* Fix CRTM fix file installation

* Use -qopenmp instead of -openmp

* Use --no-check-certificate on Orion because of OS cert issue

* Detect libdir dynamically for bufr using lua in the modulefile

* Move UPP build into its own build script and support makefile build

Commit:
Fix UPP build by setting myFC var

Commit:
Remve MPI build that was accidentally introduced

kgerheiser committed:
Merge pull request #397 from kgerheiser/bugfix/gfsv16-build

Bugfixes for GFSv16 build

natalie-perlin committed:
Add Intel CentOS7 config (#394)

* Adapted to linux-centos-intel

General Linux-CentOS (7), Intel compilers and MPI

* Update build_libtiff.sh

* Update build_proj.sh

kgerheiser committed:
Fix NCO build on Hera 

Disable ncap2 which requires Antlr libraries. This disables the C++ NCO libraries which we don't use.

kgerheiser committed:
Merge pull request #399 from NOAA-EMC/bugfix/nco-ncap2

Fix NCO build on Hera

Hang-Lei-NOAA committed:
update crtm #400

gspetro-NOAA committed:
HPC-Stack Documentation (#385)

Add Sphinx docs to hpc-stack

kgerheiser committed:
Merge branch 'develop' into feature/updatecrtm

Hang-Lei-NOAA committed:
Merge pull request #401 from Hang-Lei-NOAA/feature/updatecrtm

update crtm #400

kgerheiser committed:
Use ipolates=1 for wgrib2 (#404)

wgrib2 2.0.7 does not support ip=3

climbfuji committed:
Update Cheyenne configurations for Intel and GNU (#406)

kgerheiser committed:
Add option to build ESMF directly with tag and option for install_as (#417)

ESMF tagging scheme has changed recently, instead of being ESMF_x_y_z_beta_snapshot_a_b, it's vx.y.zbab

And set option in MAPL stack file with esmf_version to control which version it's built with.

kgerheiser committed:
Change load, try_load, and prereq to depends_on

depends_on is the correct behavior in most circumstances. If the dependent module is already loaded it will use that, but if not will try to load it. This way users can mix and match their own versions without LMod changing their selection.

depends_on combines the functionality of load/preqreq in a single step.

In some cases such as MAPL, compiler/MPI meta modules, and HDF5 with szip load/try_load where the module is has to exist or might not exist load/try_load is the correct behavior.

kgerheiser committed:
Merge pull request #418 from kgerheiser/module_depends_on

Change load, try_load, and prereq to depends_on

natalie-perlin committed:
Feature/mac gnu (#413)

Added functionality for building on MacOS X (both M1/arm64 and Intel/x86_64)

kgerheiser committed:
Remove climbfuji from install requests

Position on Cheyenne has been taken over by EPIC and @jkbk2004

kgerheiser committed:
Update README.md

kgerheiser committed:
Merge pull request #422 from NOAA-EMC/update-maintainers

Update maintainers

natalie-perlin committed:
Update macOS GNU build

* Initial changes for mac M1 arm64 architecture, GNU compilers gcc@11, openmpi-4.1.2

* added mac M1 functionality

* naming convention change to mac_m1

* libs/build_madis.sh, build_mapl.sh - remove flags

* added stack_mac_m1.yaml

* Two separate configurations for Macs: x86_64 and mac_m1, config_<machine>.sh and stack_<machine>.yaml files

* Fix configure file for mac m86_64

Co-authored-by: Natalie Perlin <Natalie@Natalies-MacBook-Air.local>
Co-authored-by: kgerheiser <kgerheiser@icloud.com>

gspetro-NOAA committed:
Documentation for Installation of hpc-stack on MacOS (#425)

Hang-Lei-NOAA committed:
adding grib_util/1.2.4

Hang-Lei-NOAA committed:
update stack files

gspetro-NOAA committed:
Make commands portable to older macs (#429)

* change commands to work for older bash/Macs

Co-authored-by: gspetro <gillian.s.petro@gmail.com>
Co-authored-by: Kyle Gerheiser <3209794+kgerheiser@users.noreply.github.com>

kgerheiser committed:
Merge branch 'develop' into feature/updategributil

Hang-Lei-NOAA committed:
Merge pull request #431 from Hang-Lei-NOAA/feature/updategributil

adding grib_util/1.2.4

kgerheiser committed:
Update FMS and nccmp versions (#433)

* Add FMS 2022.01
* Add nccmp 1.9.1.0

gspetro-NOAA committed:
Reformat mac-install.rst for inclusion in SRW submodule (#434)

kgerheiser committed:
Update NCIO to v1.1.0 (#442)

Commit:
Use Lmod functionality to save and restore module collections

Save/restore Lmod module collections; prevent reloading system default model versions by quering  construct

natalie-perlin committed:
Updated build_esmf.sh, correction in variable name

correction of the variable to form a name of the module collection, 
Line 194 in build_esmf.sh. (current <-- previous)
$name/$version_install  <--  $module/$version_install

natalie-perlin committed:
Updated build_esmf.sh

Update to build_esmf.sh, line 195,  correct Lmod syntax:

'module save hpc-$modpath-esmf'

gspetro-NOAA committed:
Minor documentation updates for installation on MacOS (#446)

natalie-perlin committed:
corrections to two libs/build_<>.sh for reloading Lmod module collections

natalie-perlin committed:
Merge branch 'feature/cheyenne' of github.com:NOAA-EPIC/hpc-stack into feature/cheyenne

natalie-perlin committed:
Merge branch 'develop' into feature/cheyenne

natalie-perlin committed:
Updates to use Lmod module collection functionality

natalie-perlin committed:
Merge branch 'feature/cheyenne' of github.com:NOAA-EPIC/hpc-stack into feature/cheyenne

natalie-perlin committed:
Updated HPC-stack documentation for the MacOS build

kgerheiser commented:
What do you want to do with this PR @natalie-perlin? I don't know what this `feature/cheyenne` branch is.

natalie-perlin committed:
HPC-stack documentation updates with the example for Linux Ubuntu20.04

kgerheiser committed:
Merge pull request #461 from NOAA-EPIC/docs-linux

HPC-stack documentation updates with the example for Linux Ubuntu20.04

natalie-perlin committed:
Merge branch 'develop' into docs

natalie-perlin committed:
Merge pull request #458 from NOAA-EPIC/docs

Updated documentation for MacOS build of hpc-stack

aerorahul committed:
add ncdiag to the stack (#466)

* add ncdiag to the stack

kgerheiser committed:
Mapl v2.22.0 (#473)

* Update MAPL and gftl-shared

* Add -DUSE_EXTDATA2G=OFF option to MAPL

t-brown committed:
Fix UPP makefile build when not using modules

When building UPP without modules it fails as it is unable
to find the dependent FORTRAN module files.

Co-authored-by: Kyle Gerheiser <3209794+kgerheiser@users.noreply.github.com>

natalie-perlin committed:
Merge branch 'develop' into feature/cheyenne
